source/common/http/header_utility.cc

```diff
@@ -225,6 +226,18 @@ bool HeaderUtility::isConnectResponse(const RequestHeaderMap* request_headers,
              Http::Code::OK;
 }
 
+bool HeaderUtility::isCapsuleProtocol(const RequestOrResponseHeaderMap& headers) {
+  Http::HeaderMap::GetResult capsule_protocol =
+      headers.get(Envoy::Http::LowerCaseString("Capsule-Protocol"));
+  if (capsule_protocol.empty()) {
+    return false;
+  }
+  // Parses the header value and extracts the boolean value ignoring parameters.
+  absl::optional<quiche::structured_headers::ParameterizedItem> header_item =
+      quiche::structured_headers::ParseItem(capsule_protocol[0]->value().getStringView());
```

Comment:
if there's multiple header entries why do we only look at the first one?

Reply:
I think we expect there is only one Capsule-Protocol header entry if it's present. Is there a relevant HTTP specification related to this case? Any thoughts @DavidSchinazi?

Reply:
Should it be an error if we expect only one header value but there are multiple?

Reply:
I'd like a comment inline on why we return false here.
It's weird to me that if there's 2 headers both of which indicate support we'd return false but I'll let @DavidSchinazi and @RyanTheOptimist make the call on what we do, and I will largely ask that we explicitly comment the reasoning inline.

I could see rejecting the request, but accepting the request and ignoring the header is not the behavior I'd default to.

Reply:
The spec says that if multiple headers are included, then they MUST be all ignored. That's because the following entries are equivalent in terms of HTTP Semantics:
```
Capsule-Protocol: ?1
Capsule-Protocol: ?1
```

```
Capsule-Protocol: ?1, ?1
```
And the second is parsed as a list and therefore dropped because of an unexpected type.

https://www.rfc-editor.org/rfc/rfc9297#name-the-capsule-protocol-header